<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1161</width>
    <height>842</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">border:none;</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">background-color:#0b192c;
</string>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">
QTabBar::tab {
min-width: 220px;
min-height:50px;
color:#f5f1e9;
background-color:#3d4f61;
margin-right:25px;
font-weight:bold;
font-size:25px;
border-radius:15px;


}

QTabBar::tab:selected {
border:none;
border-bottom:5px solid #f5f1e9;

}

QTabBar::tab:hover {
    background-color: #6b8299; /* Lighter shade on hover */
}</string>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="iconSize">
       <size>
        <width>30</width>
        <height>30</height>
       </size>
      </property>
      <property name="movable">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="threshold_tab">
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <attribute name="title">
        <string>Thresholding</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout" stretch="0,5,0,3,0">
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="DisplayArea" stretch="1,4,1,4,1">
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="thresh_original_img">
            <property name="styleSheet">
             <string notr="true">border:1px solid #f5f1e9;
border-radius:10px;
padding:10px;</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="scaledContents">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="thresh_output_img">
            <property name="styleSheet">
             <string notr="true">border:1px solid #f5f1e9;
border-radius:10px;
padding:10px;</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="settings_layout" stretch="1,4,1,3,1">
          <item>
           <spacer name="horizontalSpacer_6">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QVBoxLayout" name="thresholding_settings" stretch="2,2">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_12">
              <item>
               <widget class="QLabel" name="label">
                <property name="styleSheet">
                 <string notr="true">margin-left:5px;
color:#f5f1e9;
font-weight:bold;
font-size:20px;
</string>
                </property>
                <property name="text">
                 <string>thresholding method</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="thresholding_method">
                <property name="styleSheet">
                 <string notr="true">padding:15px 20px;
color:#f5f1e9;
background-color:#3d4f61;
margin-right:10px;
font-weight:bold;
font-size:20px;
border-radius:15px;</string>
                </property>
                <item>
                 <property name="text">
                  <string>Optimal</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Local</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Otsu</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Spectral</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_9"/>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QVBoxLayout" name="buttons_layout">
            <item>
             <widget class="QPushButton" name="load_image_btn">
              <property name="styleSheet">
               <string notr="true">QPushButton{
padding:15px 20px;
color:#f5f1e9;
background-color:#3d4f61;
margin-right:10px;
font-weight:bold;
font-size:20px;
border-radius:15px;
}
QPushButton:hover{
background-color:#6b8299;
}</string>
              </property>
              <property name="text">
               <string>Load Image</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="apply_btn">
              <property name="styleSheet">
               <string notr="true">QPushButton{
padding:15px 20px;
color:#f5f1e9;
background-color:#3d4f61;
margin-right:10px;
font-weight:bold;
font-size:20px;
border-radius:15px;
}
QPushButton:hover{
background-color:#6b8299;
}</string>
              </property>
              <property name="text">
               <string>Apply</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="horizontalSpacer_5">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="segmentation_tab">
       <attribute name="title">
        <string>Segmentation</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_10">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_8" stretch="0,5,0,5,0">
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout" name="display_layout" stretch="1,4,1,4,1">
            <item>
             <spacer name="horizontalSpacer_8">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="seg_original_img">
              <property name="styleSheet">
               <string notr="true">border:1px solid #f5f1e9;
border-radius:10px;
padding:10px;</string>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_7">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="seg_output_img">
              <property name="styleSheet">
               <string notr="true">border:1px solid #f5f1e9;
border-radius:10px;
padding:10px;</string>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_9">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_5">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_14" stretch="1,4,1,3,1">
            <item>
             <spacer name="horizontalSpacer_12">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_15">
              <item>
               <widget class="QLabel" name="label_2">
                <property name="styleSheet">
                 <string notr="true">margin-left:5px;
color:#f5f1e9;
font-weight:bold;
font-size:20px;
margin-right:10px;</string>
                </property>
                <property name="text">
                 <string>Method</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="seg_method_comboBox">
                <property name="styleSheet">
                 <string notr="true">padding:15px 20px;
color:#f5f1e9;
background-color:#3d4f61;
margin-right:10px;
font-weight:bold;
font-size:15px;
border-radius:15px;</string>
                </property>
                <item>
                 <property name="text">
                  <string>K-means</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Region growing</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>agglomerative</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>mean-shift</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="horizontalSpacer_10">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_11">
              <item>
               <widget class="QPushButton" name="load_btn">
                <property name="styleSheet">
                 <string notr="true">QPushButton{
padding:15px 20px;
color:#f5f1e9;
background-color:#3d4f61;
margin-right:10px;
font-weight:bold;
font-size:20px;
border-radius:15px;
}
QPushButton:hover{
background-color:#6b8299;
}</string>
                </property>
                <property name="text">
                 <string>Load Image</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="apply_btn_2">
                <property name="styleSheet">
                 <string notr="true">QPushButton{
padding:15px 20px;
color:#f5f1e9;
background-color:#3d4f61;
margin-right:10px;
font-weight:bold;
font-size:20px;
border-radius:15px;
}
QPushButton:hover{
background-color:#6b8299;
}</string>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="horizontalSpacer_11">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_6">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
